yarn-project/simulator/src/avm/opcodes/addressing_mode.ts

```diff
@@ -51,7 +51,7 @@ export class Addressing {
     for (const [i, offset] of offsets.entries()) {
       switch (this.modePerOperand[i]) {
         case AddressingMode.INDIRECT:
-          mem.checkTag(TypeTag.UINT32, offset);
+          mem.checkTag(TypeTag.UINT64, offset); // brillig word size is 64 bits
           resolved[i] = Number(mem.get(offset).toBigInt());
```

Comment:
:thinking: we shouldn't have to do this since AVM memory offsets are all supposed to be 32 bits

Reply:
This is a consequence of the comment above, the address space of brillig / ssa were misaligned, as long as no one is trying to use the avm this week, I can wait for that to be resolved and just stack on this pr